### ISSUE TITLE
AlgebraicField.has_assoc_Ring is False

### ISSUE THREAD

**Issue by skirpichev** [sympy/sympy#14402]
Unless I misunderstand something, this must be True and ``get_ring()`` method for AlgebraicField must point to it's ring of algebraic integers.

**Comment by jksuom** [comment]
I agree that, ideally, that should be so. But there is currently no implementation for the maximal order.
Zassenhaus has given some algorithms for its computation (Cohen: A Course in Computational Algebraic Number Theory, 6.1.8; Pohst & Zassenhaus: Algorithmic Algebraic Number Theory, 4.6) but they seem not easy to implement.

**Comment by skirpichev** [comment]
I doubt that not implemented algorithms are able to excuse wrong results.  ``get_ring()`` could raise NotImplementedError for while, that's fine.  But ``has_assoc_Ring`` class property must be fixed.

**Comment by asmeurer** [comment]
Is it specified somewhere that it should mean that? My understanding of the "associated ring" stuff in the polys domains was always that it was rather ill defined.

**Comment by skirpichev** [comment]
It was added in 622686b99.  I doubt it was later documented somewhere.
